chatbot-core/backend/app/databases/minio.py

```diff
@@ -0,0 +1,102 @@
+import os
+from minio import Minio
+from minio.error import S3Error
+from typing import BinaryIO
+
+from app.settings import Secrets
+from app.utils.logger import LoggerFactory
+
+logger = LoggerFactory().get_logger(__name__)
+
+
+class MinioConnector:
+    """
+    Minio connector class
+
+    Pattern: Singleton
+    Purpose: Create a single instance of the object storage connection
+    """
+
+    _instance = None
+    _minio_client = None
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super(MinioConnector, cls).__new__(cls)
+            cls._minio_client = cls._create_minio_client()
+        return cls._instance
+
+    @property
+    def client(self) -> Minio:
+        """
+        Get the Minio client instance
+        """
+        return self._minio_client
+
+    @classmethod
+    def get_instance(cls) -> "MinioConnector":
+        """
+        Get the instance of the MinioConnector
+        """
+        if cls._instance is None:
+            cls._instance = cls()
+        return cls._instance
+
+    @staticmethod
+    def _create_minio_client() -> Minio:
+        """
+        Create the object storage connection if there is no any existing connection
+        """
+        try:
+            return Minio(
+                endpoint=Secrets.MINIO_ENDPOINT,
+                access_key=Secrets.MINIO_ACCESS_KEY,
+                secret_key=Secrets.MINIO_SECRET_KEY,
+                secure=False,
+            )
+        except S3Error as e:
+            logger.error(f"Error initializing object storage: {e}")
+            raise
+
+    def create_bucket(self, bucket_name: str) -> None:
+        """
+        Create a bucket in object storage
+        """
+        found = self.client.bucket_exists(bucket_name=bucket_name)
+        if not found:
+            logger.info(f"Bucket {bucket_name} not found. Creating...")
+            self.client.make_bucket(bucket_name)
+        else:
+            logger.info(f"Bucket {bucket_name} already exists")
+
+    def upload_files(self, object_name: str, data: BinaryIO, bucket_name: str, length: int = None) -> None:
```

Comment:
Implemented. In testing progress ...

Reply:
Checkout my new commit [ced6424](https://github.com/TinhHoaSolutions-EzHR/chatbot/pull/50/commits/ced64248dbf812d9a9f29cb8785995dbef36de70). 

I have implemented better error handling and resource management. I cannot put the line `with contextlib.closing(data) as file_data:` in this logic because it will close the `data` object which is our uploaded document. The reason here is this document will need to be indexed into vector database as it cannot be closed.